flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/aggregate/LagAggFunction.java

```diff
@@ -44,6 +44,15 @@ public LagAggFunction(LogicalType[] valueTypes) {
                 Arrays.stream(valueTypes)
                         .map(DataTypeUtils::toInternalDataType)
                         .toArray(DataType[]::new);
+        // When the initial value has not been initialized and is not allowed to be null or
+        // default value allowed to be empty,output should be changed to allow nulls
```

Comment:
The output value can only be not null if the default input arguments include a non-null default value.